src/gebt_poc/solver.cpp

```diff
@@ -557,4 +557,67 @@ Kokkos::View<double**> CalculateStaticIterationMatrix(
     return iteration_matrix;
 }
 
+Kokkos::View<double*> ConstraintsResidualVector(
+    const Kokkos::View<double*> gen_coords, const Kokkos::View<double*> position_vector
+) {
+    auto rotation_0 = gen_alpha_solver::EulerParameterToRotationMatrix(
+        gen_alpha_solver::create_vector({gen_coords(3), gen_coords(4), gen_coords(5), gen_coords(6)})
+    );
+    auto x0 =
```

Comment:
`x0` can be a subview of `position_vector` instead of a copy:

```c++
    // Extract x0 from position vector
    auto x0 = Kokkos::subview(position_vector, Kokkos::make_pair(0, 3));
```